test/integration/auth.test.js

```diff
@@ -254,4 +256,296 @@ describe("auth", function () {
+      .get(`/auth/google/callback`)
+      .query({ code: "codeReturnedByGoogle", state: invalidRedirectUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUiUrl);
+  });
+
+  it("should redirect the google user to realdevsquad.com if invalid redirect URL provided", async function () {
+    await addUserToDBForTest(googleUserInfo[1]);
+    const invalidRedirectUrl = "invalidURL";
+    const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl")).href;
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get(`/auth/google/callback`)
+      .query({ code: "codeReturnedByGoogle", state: invalidRedirectUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUiUrl);
+  });
+
+  it("should send a cookie with JWT in the response for google user", function (done) {
+    const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl"));
+
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    chai
+      .request(app)
+      .get("/auth/google/callback")
+      .query({ code: "codeReturnedByGoogle" })
+      .redirects(0)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+
+        expect(res).to.have.status(302);
+        expect(res.headers["set-cookie"]).to.have.length(1);
+        expect(res.headers["set-cookie"][0])
+          .to.be.a("string")
+          .and.satisfy((msg) => msg.startsWith(config.get("userToken.cookieName")));
+        expect(res.headers["set-cookie"][0]).to.include("HttpOnly");
+        expect(res.headers["set-cookie"][0]).to.include("Secure");
+        expect(res.headers["set-cookie"][0]).to.include(`Domain=${rdsUiUrl.hostname}`);
+        expect(res.headers["set-cookie"][0]).to.include("SameSite=Lax");
+
+        return done();
+      });
+  });
+
+  it("should redirect the google user to login page if the user is a developer", async function () {
+    await addUserToDBForTest(googleUserInfo[3]);
+    const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl")).href;
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[2]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get("/auth/google/callback")
+      .query({ code: "codeReturnedByGoogle", state: rdsUiUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    const errorMessage = "Google login is restricted for developer role.";
+    const expectedUrl = `https://realdevsquad.com/?error=${encodeURIComponent(errorMessage)}`;
```

Comment:
Why is the error message being passed as a query param?

Reply:
The error message is being passed as a query param because this allows the frontend to easily retrieve and display it to the user as a toast notification.